tests/test_message_segmenting.py

```diff
@@ -49,25 +56,71 @@ def test_message_profiling_gsm7(self, short_gsm7_text, long_gsm7_text, byte_stri
             expected_total_segment_lengths=[106]
         )
 
-    def test_message_profiling_ucs2(self, short_ucs2_text, long_ucs2_text):
+    def test_message_profiling_ucs2(
+            self,
+            short_ucs2_text,
+            long_ucs2_text,
+            short_ucs2_text_byte_list_for_segments,
+            long_ucs2_text_byte_list_for_segments,
+            ucs2_with_surrogate_pairs,
+            ucs2_without_surrogate_pairs,
+            ucs2_with_surrogate_pairs_byte_list_for_segments,
+            ucs2_with_non_surrogate_pairs_byte_list_for_segments,
+    ):
         profiled_message = profiling.profile_message(short_ucs2_text)
         assert profiled_message["num_segments"] == 1
-        assert profiled_message["message_length"] == 14
+        assert profiled_message["message_length"] == 15
```

Comment:
Some of these changes are basically reverting what it was, before [being rewritten here](https://github.com/chrisconlon-klaviyo/sms-toolkit/commit/5e80ba60e9d5ea4ed8c379fce8e533d708e0a1bf#diff-5c1e15f9a12bc24f389e473c94186c89). After speaking with the author Vidhu (the author), The test-rewrite was just meant to make the tests pass in CI. Building the tests locally on his machine failed too, indicating the same problem.